### DISCUSSION TITLE
How to use infisical in a direnv workflow?

### DISCUSSION THREAD

**Discussion by gee-forr** [Infisical/infisical#2222] [General]
Hey there,

I'm a future [envkey refugee](https://mailchi.mp/f2ee890e5e81/envkey-wind-down) seeking a new home.

During dev it's a pain to have to prepend everything we do with `enkey-source -- my-cmd --blah --blah`. What we do instead is use [direnv](https://direnv.net/), so that as soon as we `cd` into a project's directory, we run an `envkey-source` in its default mode which simply exports all its secrets into the shell's environment.

From there, we can run whatever commands we want without having to prepend a secrets export command. I realise this is "less secure" as anything we run now has access to those secrets, but this is a fairly safe tradeoff as this is typically a practice we only do during development.

Is there a way to have infiscal export all the secrets for a project into the shell's env? I've gone through the CLI docs, but for the life of me, can't find an obvious solution?

**Comment by maidul98** [comment]
Hey @gee-forr , as i understand, you would like to be able to export secrets from infisical into your shell via the `export` keyword so that it can be picked by `direnv`. 

If that is the case, you can use the `infisical export --format=dotenv-export` command that will output a list in the following format:
`export SECRET_NAME=SECRET_VALUE`. You can then use this to feed to your shell. Is this what you were looking for? Let me know if i have misunderstood

**Comment by Alexnortung** [comment]
I have a solution for this, as I would like something similiar.

Based on [the reply](https://github.com/Infisical/infisical/discussions/2222#discussioncomment-10243314) by @maidul98.
You can add the following to your `.envrc`, which will automatically load and unload the environment variables.

```
eval $(infisical export --format=dotenv-export --silent)
```

**Reply by bbigras** [reply]
> eval $(infisical export --format=dotenv-export --silent)

Is it possible to ask the user for confirmation before exporting the secrets?

**Reply by Alexnortung** [reply]
@bbigras something like this?

```bash
read -p "Do you want to export secrets from Infisical? (y/N) " -n 1 -r
echo    # move to a new line
if [[ $REPLY =~ ^[Yy]$ ]]
then
    eval $(infisical export --format=dotenv-export --silent)
fi
```

**Reply by bbigras** [reply]
Thanks, but I meant on the infisical side. To prevent a malicious script from running `infisical export` and stealing my precious.
